web/src/engine/providers/BookmarkPlugin.ts

```diff
@@ -23,7 +25,17 @@ export class BookmarkPlugin extends MediaContainer<Bookmark> {
     }
 
     private Deserialize(serialized: BookmarkSerialized): Bookmark {
-        const parent = this.plugins.WebsitePlugins.find(plugin => plugin.Identifier === serialized.Media.ProviderID);
+        let parent = this.plugins.WebsitePlugins.find(plugin => plugin.Identifier === serialized.Media.ProviderID);
+        const dummyPluginName = `deleted_${serialized.Media.ProviderID}`;
+        if (!parent) {
```

Comment:
I think you don't have to create a full fledged plugin and also don't have to add it to the plugin list (which would expose all these dummy plugins to the user).

Try to use a more minimalistic approach, if there is no parent, just create an object that satisfies the `IMediaContainer` interface.

```javascript
_createInvalidParent(name: string, identifier, string): IMediaContainer {
    return {
        // dummy properties and empty functions for IMediaContainer
    };
}
// ...
const parent = this.plugins.WebsitePlugins.find(/*...*/) ?? this._createInvalidParent('[DELETED] ' + serialized.Media.Label, serialized.Media.ProviderID);
```

Reply:
Done, i guess. There is an exception on invalid bookmark clicking because Origin doesnt exist but it seems to be handled properly and doesnt impact program usability.